control-plane/grpc/src/operations/nexus/traits.rs

```diff
@@ -1193,3 +1349,59 @@ impl From<&dyn RemoveNexusChildInfo> for RemoveNexusChild {
         }
     }
 }
+
+/// GetRebuildRecordInfo trait for the add nexus child to be implemented by entities which want to
+/// use this operation.
+pub trait GetRebuildRecordInfo: Send + Sync + std::fmt::Debug {
+    /// Uuid of the nexus.
+    fn nexus(&self) -> NexusId;
+}
+
+impl GetRebuildRecordInfo for GetRebuildRecord {
+    fn nexus(&self) -> NexusId {
+        self.nexus.clone()
+    }
+}
+
+/// Intermediate structure that validates the conversion to RebuildHistoryRequest type
+#[derive(Debug)]
+pub struct ValidatedGetRebuildRecord {
+    nexus: NexusId,
+}
+
+impl GetRebuildRecordInfo for ValidatedGetRebuildRecord {
+    fn nexus(&self) -> NexusId {
+        self.nexus.clone()
+    }
+}
+
+impl GetRebuildRecordInfo for RebuildHistoryRequest {
+    fn nexus(&self) -> NexusId {
+        NexusId::try_from(self.uuid.clone()).unwrap()
+    }
+}
+
+impl ValidateRequestTypes for RebuildHistoryRequest {
+    type Validated = ValidatedGetRebuildRecord;
```

Comment:
Can't this be NexusId?

Reply:
What do you mean?

Reply:
Seems the only validation here is for the NexusId, so there's not much point in having a new type just containing the NexusId, but fine if you want to keep it as is.

Reply:
Sure Tiago